data-collection/data-collection/View Controllers/Map View Controller/MapViewController+Extras.swift

```diff
@@ -0,0 +1,126 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import ArcGIS
+import ArcGISToolkit
+
+private let layersExtra = "Layers"
+private let bookmarksExtra = "Bookmarks"
+
+/// Defines the Extras.
+public enum Extras : CaseIterable {
+    // Displays layer content.
+    case layers
+    // Displays bookmarks.
+    case bookmarks
+}
+
+extension Extras : CustomStringConvertible {
+    public var description: String {
+        switch self {
+        case .layers:
+            return "Layers"
+        case .bookmarks:
+            return "Bookmarks"
+        }
+    }
+}
+
+extension MapViewController {
+    func userRequestsExtras(_ barButtonItem: UIBarButtonItem?) {
+        guard mapViewMode != .disabled else {
+            return
+        }
+
+        let action = UIAlertController(title: nil, message: nil, preferredStyle: .actionSheet)
+                
+        // Present the list of extras.
+        for extra in Extras.allCases {
+            var extraAction: UIAlertAction
+            switch extra {
+            case .layers:
+                extraAction = UIAlertAction(title: extra.description, style: .default, handler: { [weak self] (action) in
+                    self?.showLayerContents(barButtonItem)
+                })
+            case .bookmarks:
+                extraAction = UIAlertAction(title: extra.description, style: .default, handler: { [weak self] (action) in
+                    self?.showBookmarks(barButtonItem)
+                })
+            }
+            
+            action.addAction(extraAction)
+        }
+        
+        action.addAction(.cancel())
+        action.popoverPresentationController?.barButtonItem = barButtonItem
+        present(action, animated: true)
+    }
+    
+    func showLayerContents(_ barButtonItem: UIBarButtonItem?) {
+        var layerContentsVC: LayerContentsViewController
+        if let existingViewController = layerContentsViewController {
+            layerContentsVC = existingViewController
+        } else {
+            // Create and configure the view controller.
+            let dataSource = DataSource(geoView: mapView)
+            layerContentsVC = TableOfContents(dataSource: dataSource)
+            
+            // Add a done button.
+            let doneButton = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(done))
+            layerContentsVC.navigationItem.leftBarButtonItem = doneButton
+            layerContentsViewController = layerContentsVC
+        }
+        
+        // Display the layerContentsVC as a popover controller.
+        layerContentsVC.modalPresentationStyle = .popover
+        layerContentsVC.popoverPresentationController?.barButtonItem = barButtonItem
+        present(layerContentsVC, animated: true)
+    }
+    
+    func showBookmarks(_ barButtonItem: UIBarButtonItem?) {
+        var bookmarksVC: BookmarksViewController
```

Comment:
This doesn't need to be mutable.

```suggestion
        let bookmarksVC: BookmarksViewController
```